src/averageAges.js

```diff
@@ -55,7 +67,36 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const filteredHasMother = (onlyWithSon !== undefined)
+    ? people.filter(person =>
+      person.sex === 'm'
+      && people.find(mother => mother.name === person.mother)
+    )
+    : people.filter(person =>
+      people.find(mother => mother.name === person.mother)
+    );
+
+  const sumOfAges = filteredHasMother.reduce((sum, person) => {
+    const ageDiff = (person.born - people.find(mother =>
+      mother.name === person.mother).born
+    );
+
+    return sum + ageDiff;
+  }, 0);
+
+  return sumOfAges / filteredHasMother.length;
+}
+
+// Helper function to calculate average age:
+
+function calculateAverageAge(filteredArray) {
+  const sumOfAges = filteredArray.reduce((sum, person) =>
+    sum + (person.died - person.born), 0
+  );
```

Comment:
```suggestion
  const sumOfAges = filteredArray.reduce(
    (sum, person) => sum + (person.died - person.born), 0
  );
```

Reply:
Or you can wrap arrow function

Reply:
hope I fixed it everywhere)
these wrappings are real hell)